### PR TITLE
more typos

### PR DESCRIPTION
**Pull request by askeblad**
Cabal-QuickCheck/src/Test/QuickCheck/GenericArbitrary.hs
Cabal-described/src/Distribution/Utils/CharSet.hs
Cabal-syntax/README.md
Cabal-syntax/src/Distribution/Compat/CharParsing.hs
Cabal-syntax/src/Distribution/Compat/Newtype.hs
Cabal-syntax/src/Distribution/FieldGrammar.hs
Cabal-syntax/src/Distribution/Fields.hs
Cabal-syntax/src/Distribution/Fields/Pretty.hs
Cabal-syntax/src/Distribution/PackageDescription/Parsec.hs
Cabal-syntax/src/Distribution/System.hs
Cabal-syntax/src/Distribution/Types/BuildInfo.hs
Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs
Cabal-syntax/src/Distribution/Types/VersionInterval.hs
Cabal-syntax/src/Distribution/Types/VersionInterval/Legacy.hs
Cabal-syntax/src/Distribution/Utils/Structured.hs
Cabal-tests/Cabal-tests.cabal
Cabal-tests/tests/Test/Laws.hs
Cabal-tests/tests/cbits/rpmvercmp.c
Cabal/ChangeLog.md
Cabal/src/Distribution/PackageDescription/Check.hs
Cabal/src/Distribution/Simple/Build/Macros.hs
Cabal/src/Distribution/Simple/BuildToolDepends.hs
Cabal/src/Distribution/Simple/Configure.hs
Cabal/src/Distribution/Simple/Utils.hs
Cabal/src/Distribution/Utils/NubList.hs
Makefile
cabal-dev-scripts/src/GenUtils.hs
cabal-install-solver/src/Distribution/Solver/Modular/Dependency.hs
cabal-install-solver/src/Distribution/Solver/Modular/Linking.hs
cabal-install-solver/src/Distribution/Solver/Modular/MessageUtils.hs
cabal-install-solver/src/Distribution/Solver/Types/PackageIndex.hs
cabal-install-solver/tests/UnitTests/Distribution/Solver/Modular/MessageUtils.hs
cabal-install/changelog
cabal-install/src/Distribution/Client/Check.hs
cabal-install/src/Distribution/Client/CmdErrorMessages.hs
cabal-install/src/Distribution/Client/Compat/Semaphore.hs
cabal-install/src/Distribution/Client/Fetch.hs
cabal-install/src/Distribution/Client/HashValue.hs
cabal-install/src/Distribution/Client/ProjectConfig.hs
cabal-install/src/Distribution/Client/ProjectPlanning.hs
cabal-install/src/Distribution/Client/RebuildMonad.hs
cabal-install/src/Distribution/Client/Setup.hs
cabal-install/src/Distribution/Client/SrcDist.hs
cabal-install/src/Distribution/Client/TargetSelector.hs
cabal-install/src/Distribution/Client/Types/Repo.hs
cabal-install/tests/UnitTests/Distribution/Client/VCS.hs
cabal-install/tests/UnitTests/Distribution/Solver/Modular/Solver.hs
cabal-testsuite/PackageTests/CCompilerOverride/setup.test.hs
cabal-testsuite/PackageTests/DeterministicAr/setup-default-ar.test.hs
cabal-testsuite/PackageTests/DeterministicAr/setup-old-ar-without-at-args.test.hs
cabal-testsuite/PackageTests/Regression/T5309/T5309.cabal
cabal-testsuite/PackageTests/Regression/T5309/lib/Data/TCM/Memoized/FFI.hsc
cabal-testsuite/cabal-testsuite.cabal
doc/cabal-project.rst
doc/developing-packages.rst
doc/file-format-changelog.rst
doc/getting-started.rst
editors/vim/example.cabal
release-notes/Cabal-3.6.1.0.md
release-notes/cabal-install-3.4.0.0.md
stack.yaml
validate.sh

<hr>

@Mikolaj 

any opinion on the below

Cabal/ChangeLog.md

```
  * Static linking
    * Add `--enable-executable-static` flag for building fully
      static executables (GHC's normal "statish" linking links
      Haskell libraries statically, but libc and system dependencies
      dynamically). This new flag links everything statically.
```

### CODE REVIEW

**Comment by askeblad** [comment]
a lot of files

another typo `appropiate`

maybe a Member can fix

.github/workflows/validate.yml

```diff
-         # We need to build an array dynamically to inject the appropiate env var in a previous job,
+         # We need to build an array dynamically to inject the appropriate env var in a previous job,
```

**Comment by askeblad** [comment]
@andreabedini 

care to review

File: Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs [hunk 1]
```diff
@@ -20,7 +20,7 @@ import qualified Text.PrettyPrint                as Disp
 -- @since 3.0.0.0
 --
 data LibraryVisibility
-      -- | Can be depenendent from other packages
+      -- | Can be independent from other packages
```
Comment (fgaz) [inline Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs:23]: Can be depended on by other packages (is this correct english?)

**Review by fgaz — CHANGES_REQUESTED** [review]

File: Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs (hunk 1 above)
Comment (askeblad) [reply Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs:23]: since 3.0.0.0 

presumably the code is 'trustworthy"

so if the code can be relied on/upon by other packages

possibly

`      -- | Can be relied on by other packages`

Comment (askeblad) [reply Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs:23]: do you mean

`      -- | API exported for use by other packages`

**Comment by andreabedini** [comment]
Thanks for the PR @askeblad. I think @fgaz has found the only tricky bit, the rest looks ok to me. There's a sentence which still sounds confusing but it already did before, I am happy if we just fix the typo in the meantime.

**Comment by askeblad** [comment]
thanks for the review

> @fgaz has found the only tricky bit
> 
> a sentence which still sounds confusing

is the latter the same as the former

if not care to identify the latter in the source code

**Comment by askeblad** [comment]
quoted string isn't an English word

Cabal/ChangeLog.md

`GHC's normal "statish" linking links Haskell libraries statically`

**Comment by Mikolaj** [comment]
> quoted string isn't an English word
> 
> Cabal/ChangeLog.md
> 
> `GHC's normal "statish" linking links Haskell libraries statically`

That's fine IMHO. That's a new word formation with a reasonably clear meaning.

**Comment by andreabedini** [comment]
@askeblad Yes, I should have linked it but I was on my phone.

> -- Test unneeded version bound on internal build-tools deps

before it was

> -- Test unneed version bound on internal build-tools deps

Which (IMHO) is not more comprehensible.

It appears twice in the test suite

https://github.com/haskell/cabal/pull/8099/files#diff-7c514a936559ada2fb65d385273c9e334884733fc4e0ddf12d0e92dcdd6222ceR2

https://github.com/haskell/cabal/pull/8099/files#diff-f371e48c94c4ee6eaaab13fbf27f52999295c64b6c7dabd2d9fcedac44b7cddaR2

**Comment by askeblad** [comment]
@andreabedini 

FWIW. I had from the context a difficult time extracting meaning from that comment.

Likewise, with the comment line pointed out by @fgaz.

**Comment by askeblad** [comment]
@Mikolaj 

> "statish"

an Internet search for the quoted keyword returned it's use in slang

**Comment by andreabedini** [comment]
I'd interpret the word "statish" in 

> GHC's normal "statish" linking links Haskell libraries statically, but libc and system dependencies dynamically

as meaning "almost static".

**Comment by askeblad** [comment]
suppose this is as good a place as any to post this

run in uxterm

`grep -o -E -r "\w+('\w+)*" cabal | sort -u -f > output`

`cut -d : -f2 output | sort -u -f > words`

any scripting language preferences to reduce the number of lines in `words` file by

1. removing alphanumeric and numeric strings
2. removing camel case strings and strings with underscores

File: Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs (hunk 1 above)
Comment (emilypi) [reply Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs:23]: `Can be used as a dependency for other packages` is closest imo. The original fix is actually quite wrong. Were you just running a spell checker + replacement recursively on the project?

File: cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs [hunk 2]
```diff
@@ -25,7 +25,7 @@ import Distribution.Package               (getHSLibraryName)
 import Distribution.Simple.Compiler       (compilerId)
 import Distribution.Simple.LocalBuildInfo (LocalBuildInfo, compiler, localUnitId)
 
--- Almost a copypasta of Distribution.Simple.Program.Ar.wipeMetadata
+-- Almost a copy/paste of Distribution.Simple.Program.Ar.wipeMetadata
```
Comment (emilypi) [inline cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs:28]: Revert. this is fine and is common internet vernacular.

File: changelog.d/pr-8023 [hunk 3]
```diff
@@ -6,6 +6,6 @@ description: {
 
 - Make builds that use experimental Cabal language features less noisy. At -v1
   (normal) we show just first instance of use of experimental cabal language
-features, along with count of further occurences in the same file.
+features, along with count of further occurrences in the same file.
 
 }
```
Comment (emilypi) [inline changelog.d/pr-8023:11]: Please do not touch past changelogs

File: Cabal-syntax/src/Distribution/Fields/Pretty.hs [hunk 4]
```diff
@@ -59,7 +59,7 @@ showFields rann = showFields' rann (const id) 4
 -- | 'showFields' with user specified indentation.
 showFields'
   :: (ann -> CommentPosition)
-     -- ^ Convert an annotation to lined to preceed the field or section.
+     -- ^ Convert an annotation to lined to precede the field or section.
```
Comment (emilypi) [inline Cabal-syntax/src/Distribution/Fields/Pretty.hs:62]: Grammar should be fixed, not just spelling. 

Suggesting 

```suggestion
     -- ^ Convert an annotation to a line preceding the field or section.
```

**Review by emilypi — COMMENTED** [review]
Comments

File: cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs (hunk 2 above)
Comment (andreabedini) [reply cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs:28]: Personally, I'd prefer "... a copy and paste of ..."

Comment (askeblad) [reply cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs:28]: > Personally, I'd prefer "... a copy and paste of ..."

same with me

**Comment by askeblad** [comment]
1st result
(quoted keyword)

[statish - Urban Dictionary](https://www.urbandictionary.com/define.php?term=statish)

**Comment by askeblad** [comment]
@ondrap

how would you rewrite the first comment

Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs

```haskell
data LibraryVisibility
      -- | Can be depenendent from other packages
    = LibraryVisibilityPublic
      -- | Internal library, default
    | LibraryVisibilityPrivate
    deriving (Generic, Show, Read, Eq, Typeable, Data)
```
and these comments

cabal-testsuite/PackageTests/InternalVersions/BuildDependsExtra/setup.test.hs

```haskell
import Test.Cabal.Prelude
-- Test unneed version bound on internal build-tools deps
main = setupAndCabalTest . expectBroken 7470 $ do
    setup' "configure" []
    assertOutputContains "extraneous version range"
        =<< setup' "sdist" []
```

cabal-testsuite/PackageTests/InternalVersions/BuildToolsExtra/setup.test.hs

```haskell
import Test.Cabal.Prelude
-- Test unneed version bound on internal build-tools deps
main = setupAndCabalTest . expectBroken 7470 $ do
    setup' "configure" []
    assertOutputContains "extraneous version range"
        =<< setup' "sdist" []
```

**Comment by askeblad** [comment]
https://github.com/haskell/cabal/pull/8099#issuecomment-1100745342

before this closes will a Member fix
`.github/workflows/validate.yml`

File: Cabal-syntax/src/Distribution/Fields/Pretty.hs (hunk 4 above)
Comment (askeblad) [reply Cabal-syntax/src/Distribution/Fields/Pretty.hs:62]: I think it's clear enough to me, but I wouldn't object to the suggestion.

**Comment by Mikolaj** [comment]
> [#8099 (comment)](https://github.com/haskell/cabal/pull/8099#issuecomment-1100745342)
> 
> before this closes will a Member fix `.github/workflows/validate.yml`

I hope it's fixed now. Could you rebase?

File: Cabal-syntax/src/Distribution/Fields/Pretty.hs (hunk 4 above)
Comment (emilypi) [reply Cabal-syntax/src/Distribution/Fields/Pretty.hs:62]: It doesn't make sense with just the spelling change?

File: cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs (hunk 2 above)
Comment (emilypi) [reply cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs:28]: But the question is not what you both prefer, it's whether or not the original text was clear in its intent and ostensible correctness, and *not touching unless necessary*. We want meaningful changes, not just change for the sake of it. This to me does not meet the bar for meaningful change.

File: changelog.d/pr-8023 (hunk 3 above)
Comment (askeblad) [reply changelog.d/pr-8023]: https://github.com/haskell/ThreadScope/pull/123#issuecomment-1105604564

**Comment by askeblad** [comment]
@Mikolaj

from a fork to another Organization than this PR is from

```
$ git clone -q https://github.com/varehus/cabal.git
$ grep appropiate cabal/.github/workflows/validate.yml
        # We need to build an array dynamically to inject the appropiate env var in a previous job,
$ 
```

**Comment by Mikolaj** [comment]
@askeblad; I miss your point. They have typos that we don't have? So what?

**Comment by askeblad** [comment]
> Could you rebase?

doc/cabal-package.rst

someone else can fix the typo

File: doc/cabal-package.rst [hunk 5]
```diff
@@ -1037,39 +1037,36 @@ The command writes the selected version for all dependencies to the
 the dependency versions specified in it.
 
 Generating dependency version bounds
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+""""""""""""""""""""""""""""""""""""
 
 Cabal also has the ability to suggest dependency version bounds that
-conform to the `Package Versioning Policy`_, which is
+conform to `Package Versioning Policy`_, which is
```
Comment (emilypi) [inline doc/cabal-package.rst:1043]: undo this - it is grammatically incorrect

File: doc/cabal-package.rst [hunk 6]
```diff
@@ -1037,39 +1037,36 @@ The command writes the selected version for all dependencies to the
...
 
 ::
 
     build-depends:
-      base          >= 4.15.0 && < 4.16,
-      mtl           >= 2.2.2 && < 2.3,
-      transformers  >= 0.5.6 && < 0.6,
-
+      foo >= 0.5.2 && < 0.6
+      bar >= 1.1 && < 1.2
```
Comment (emilypi) [inline doc/cabal-package.rst:1066]: The semantics of this section have changed, and are now less informative than before. Please revert. The point of this section is to show that bounds-less dependencies in the presence of a `cabal gen-bounds` run produces a set of bounds for those deps.

File: doc/cabal-package.rst [hunk 7]
```diff
@@ -1119,8 +1154,6 @@ Examples:
     Outdated dependencies:
     HUnit ==1.3.1.1 (latest: 1.3.1.2)
 
-See `the command documentation <cabal-commands.html#cabal-outdated>`__ for a
-list of available flags.
```
Comment (emilypi) [inline doc/cabal-package.rst:1123]: Why was this deleted? If you inlined it above, did it not exist before? If it existed before, then this is an extraneous change and should be reverted

**Review by emilypi — CHANGES_REQUESTED** [review]

File: doc/cabal-package.rst (hunk 7 above)
Comment (askeblad) [reply doc/cabal-package.rst:1123]: doc/cabal-package.rst

subsequent to my typo correction

the upstream file was changed

could not rebase so reverted that change

still could not rebase

so used upstream one

File: Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs (hunk 1 above)
Comment (askeblad) [reply Cabal-syntax/src/Distribution/Types/LibraryVisibility.hs]: the Member comment was committed

**Comment by askeblad** [comment]
@Mikolaj re-reviews unnecessary

your prevoius 2 comments could be deleted

**Comment by askeblad** [comment]
@0intro
@rsc

[Plan 9](https://gitlab.haskell.org/ghc/ghc/-/issues/21246) (GHC's GitLab instance - Ghost User)

**Comment by emilypi** [comment]
Who are all these people you keep pinging?

File: cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs (hunk 2 above)
Comment (andreabedini) [reply cabal-testsuite/src/Test/Cabal/CheckArMetadata.hs]: ok, fair enough. In the context of the PR, this change looked appropriate to me. Happy to reconsider if you feel strongly against it.

File: cabal-install/src/Distribution/Client/Fetch.hs [hunk 8]
```diff
@@ -57,7 +57,7 @@ import Distribution.System
 -- * support tarball URLs via ad-hoc download cache (or in -o mode?)
 -- * suggest using --no-deps, unpack or fetch -o if deps cannot be satisfied
 -- * Port various flags from install:
---   * --updage-dependencies
+--   * --update-dependencies
```
Comment (fgaz) [inline cabal-install/src/Distribution/Client/Fetch.hs:60]: It's `upgrade`!

File: Cabal-syntax/src/Distribution/Fields/Pretty.hs [hunk 9]
```diff
@@ -59,7 +59,7 @@ showFields rann = showFields' rann (const id) 4
 -- | 'showFields' with user specified indentation.
 showFields'
   :: (ann -> CommentPosition)
-     -- ^ Convert an annotation to lined to preceed the field or section.
+     -- ^ Convert an annotation to a line preceding the field or section.
```
Comment (fgaz) [inline Cabal-syntax/src/Distribution/Fields/Pretty.hs:62]: This doesn't look like what that function is supposed to do

**Review by fgaz — CHANGES_REQUESTED** [review]

**Comment by fgaz** [comment]
A this point it's clear that [this pull request is automated](https://github.com/haskell/cabal/pull/8099#discussion_r854449228). I'm not necessarily against that (I even did one myself, with hlint), but _every change_ has to be checked manually before submitting the patch!

Since reviewing this is consuming a lot of maintainers' time, I'm marking it as draft. @askeblad please check that every correction is semantically correct, and not only grammatically, and only then mark it as ready for review

File: Cabal-syntax/src/Distribution/Fields/Pretty.hs (hunk 9 above)
Comment (emilypi) [reply Cabal-syntax/src/Distribution/Fields/Pretty.hs:62]: proposing "Convert a comment annotation to a line before or after the field or section"

**Comment by askeblad** [comment]
zachowanie niezgodne z przyjętymi zwyczajami towarzyskimi
